internal/services/servicebus/servicebus_namespace_customer_managed_key_resource.go

```diff
@@ -0,0 +1,265 @@
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ServiceBus.NamespacesClient
+			var cmk ServiceBusNamespaceCustomerManagedKeyModel
+
+			if err := metadata.Decode(&cmk); err != nil {
+				return err
+			}
+
+			id, err := namespaces.ParseNamespaceID(cmk.NamespaceID)
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("%s was not found", *id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` is nil", *id)
+			}
+
+			if resp.Model.Properties != nil && resp.Model.Properties.Encryption != nil && resp.Model.Properties.Encryption.KeyVaultProperties != nil && len(*resp.Model.Properties.Encryption.KeyVaultProperties) > 0 {
+				return metadata.ResourceRequiresImport(r.ResourceType(), *id)
+			}
+
+			payload := resp.Model
+
+			keyId, err := keyVaultParse.ParseOptionallyVersionedNestedItemID(cmk.KeyVaultKeyID)
+			if err != nil {
+				return err
+			}
+
+			payload.Properties.Encryption = &namespaces.Encryption{
+				RequireInfrastructureEncryption: pointer.To(cmk.InfrastructureEncryptionEnabled),
+				KeySource:                       pointer.To(namespaces.KeySourceMicrosoftPointKeyVault),
+				KeyVaultProperties: &[]namespaces.KeyVaultProperties{
+					{
+						KeyName:     pointer.To(keyId.Name),
+						KeyVersion:  pointer.To(keyId.Version),
+						KeyVaultUri: pointer.To(keyId.KeyVaultBaseUrl),
+					},
+				},
+			}
+
+			if cmk.IdentityID != "" {
+				identityId, err := commonids.ParseUserAssignedIdentityID(cmk.IdentityID)
+				if err != nil {
+					return err
+				}
+				(*payload.Properties.Encryption.KeyVaultProperties)[0].Identity = &namespaces.UserAssignedIdentityProperties{
+					UserAssignedIdentity: pointer.To(identityId.ID()),
+				}
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *payload); err != nil {
+				return fmt.Errorf("creating Customer Managed Key for %s: %+v", *id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ServiceBusNamespaceCustomerManagedKeyResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ServiceBus.NamespacesClient
+
+			id, err := namespaces.ParseNamespaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			var state ServiceBusNamespaceCustomerManagedKeyModel
+			state.NamespaceID = id.ID()
+
+			if props := resp.Model.Properties; props != nil && props.Encryption != nil {
```

Comment:
We should also nil check model here:
```suggestion
			if model := resp.Model; model != nil {
				if props := model.Properties; props != nil && props.Encryption != nil {
				...
			}
```